tests/kernel/mem_pool/mem_pool_concept/testcase.yaml

```diff
@@ -1,3 +1,4 @@
 tests:
   kernel.memory_pool:
     tags: kernel mem_pool
+    filter: not CONFIG_SYS_HEAP_MEMPOOL_COMPAT
```

Comment:
Surely there's some test case in this that can pass with the new code.  Probably best to identify the specific set of test cases that fail instead of disabling whole tests.